test/k8s/res_limits_test.go

```diff
@@ -0,0 +1,487 @@
+      - Sidecar
+  from:
+    - targetRef:
+        kind: Mesh
+      default:
+        local:
+          http:
+            requestRate:
+              num: 10000
+              interval: 1s
+            onRateLimit:
+              status: 429
+`
+		Expect(cluster.Install(YamlK8s(policy))).To(Succeed())
+	})
+
+	Context("load", func() {
+		var cpu int
+		var memory int
+		var numServices = 5
+		var instancesPerService = 1
+		var goMemLimitAdded = false
+
+		adjustResource := func(miliCPU, memMega int, addGoMemLimitEnv bool) {
+			Logf("adjusting control plane resource limits to cpu %dm, memory %dMi\n", miliCPU, memMega)
+
+			patchJson := []string{
+				fmt.Sprintf(`{"op": "replace", "path": "/spec/template/spec/containers/0/resources/requests/cpu", "value": "%dm"}`, miliCPU),
+				fmt.Sprintf(`{"op": "replace", "path": "/spec/template/spec/containers/0/resources/requests/memory", "value": "%dMi"}`, memMega),
+				fmt.Sprintf(`{"op": "replace", "path": "/spec/template/spec/containers/0/resources/limits/cpu", "value": "%dm"}`, miliCPU),
+				fmt.Sprintf(`{"op": "replace", "path": "/spec/template/spec/containers/0/resources/limits/memory", "value": "%dMi"}`, memMega),
+			}
+
+			if addGoMemLimitEnv {
+				// set 90% of the container memory as GOMEMLIMIT, the remaining 10% is for the rest of the container
+				memLimit := int(float64(memMega) * 0.9)
+				if memLimit < 10 {
+					memLimit = 10
+				}
+
+				if !goMemLimitAdded {
+					patchJson = append(patchJson, fmt.Sprintf(`{"op": "add", "path": "/spec/template/spec/containers/0/env/-", "value": {"name": "GOMEMLIMIT", "value":"%dMiB"}}`, memLimit))
+					goMemLimitAdded = true
+				} else {
+					//  get the existing env array, and update the existing GOMEMLIMIT
+					out, err := k8s.RunKubectlAndGetOutputE(
+						cluster.GetTesting(),
+						cluster.GetKubectlOptions(Config.KumaNamespace),
+						"get", "deployment", Config.KumaServiceName,
+						"-o=jsonpath='{.spec.template.spec.containers[0].env}'")
+					Expect(err).ToNot(HaveOccurred())
+					var jsonEnvArray []map[string]interface{}
+					err = yaml.Unmarshal([]byte(out), &jsonEnvArray)
+					Expect(err).ToNot(HaveOccurred())
+					var idx = -1
+					for i, m := range jsonEnvArray {
+						if m["name"] == "GOMEMLIMIT" {
+							idx = i
+							break
+						}
+					}
+
+					op := "replace"
+					idxStr := fmt.Sprintf("%d", idx)
+					if idx == -1 {
+						op = "add"
+						idxStr = "-"
+					}
+
+					patchJson = append(patchJson, fmt.Sprintf(`{"op": "%s", "path": "/spec/template/spec/containers/0/env/%s", "value": {"name": "GOMEMLIMIT", "value":"%dMiB"}}`,
+						op, idxStr, memLimit))
+				}
+			}
+
+			err := k8s.RunKubectlE(
+				cluster.GetTesting(),
+				cluster.GetKubectlOptions(Config.KumaNamespace),
+				"patch", "deployment", Config.KumaServiceName,
+				"--type=json",
+				"--patch", fmt.Sprintf("[%s]", strings.Join(patchJson, ",")),
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			time.Sleep(2 * time.Second)
```

Comment:
Why do we sleep? Can status deployment not be available? If so, wouldn't we need Eventually?